app/controllers/users_controller.rb

```diff
@@ -34,6 +34,25 @@ def update
     end
   end
 
+  def promote
+    users = User.where.not(staff: true)
+    @users = []
+    users.each do |user|
+      @users << "#{user.full_name} #{user.spire}"
+    end
```

Comment:
That's true. I was thinking something more like

```ruby
@users = User.where.not(staff: true)
             .pluck(:first_name, :last_name, :spire)
             .map do |first_name, last_name, spire|
               "#{first_name} #{last_name} (#{spire})"
             end
```

`pluck` is good when you only want some attributes of users, and not to call any methods on them. You can them grab just those data attributes and iterate over them, which has two advantages:

1. Only a single DB query.
2. A much lighter data structure over which to iterate.